src/java.base/share/classes/sun/security/x509/X509CertImpl.java

```diff
@@ -1584,6 +1584,17 @@ public int getBasicConstraints() {
                     throw new RuntimeException("name cannot be encoded", ioe);
                 }
                 nameEntry.add(derOut.toByteArray());
+                if (name.getType() == GeneralNameInterface.NAME_ANY
+                        && name instanceof OtherName oname) {
+                    nameEntry.add(oname.getOID().toString());
+                    byte[] nameValue = oname.getNameValue();
+                    try {
+                        String v = new DerValue(nameValue).getAsString();
+                        nameEntry.add(v == null ? nameValue : v);
+                    } catch (IOException ioe) {
```

Comment:
Attention, this catch block will hide invalid ASN.1 encoding of the other name element from:
* sun.security.util.DerValue.init(boolean, InputStream, boolean)
* sun.security.util.DerValue.getIA5String()

Other blocks throw:
```
throw new CertificateException("Unable to parse DER value of SAN:otherName", ioe);
```

Do you really intend to hide an encoding error int the cert from the user?

Reply:
Up to debate. Other blocks in `makeAltNames` throw `RuntimeException`.

Reply:
But in this case, we still have a place to provide the raw bytes. Maybe that's better? Or you'd rather be sure that one particular otherName should always have a string there and there's no need to do an `instanceof` check?

Reply:
I have thought about this actually. Now the parse is free of any semantics, which is naive. Actually, you need a list of wellknown OIDs to know the target tag type to perform the conversion. E.g., you know that MS UPN must be UTF8String, if not this is an error. If you don't know the OID, don't touch it.
OpenSSL knows the semantics and decodes it, otherwise don't touch it and leave it: https://github.com/openssl/openssl/blob/317acac5cc0a2cb31bc4b91353c2b752a3989d8a/crypto/x509/v3_san.c#L113-L120

Reply:
Maybe adopt the list of OpenSSL, otherwise return byte array?

Reply:
@wangweij So you have selected to really swallow the exception here?

Reply:
Yes. I don't want to let the method fail. Since `instanceof String` should be called anyway the caller can decide how to fail.

Reply:
So you leave it to the user to read the ASN.1 value and fail if the encoding is incorrect instead of throwing a `Ceritificate..Exception` although other GeneralNames do?

Reply:
I want to minimize behavior change and I'm leaving them to notice there's no string there and fail.

Reply:
For other known names, you either return a parsed name or fail. For `otherName`, you already have the raw data and this further parsed name is a bonus. I just don't like a problem in getting the bonus to ruin the original benefit. Also, I think the caller will have to check the length and the type anyway so this will not be an extra burden. Besides, I always feel that `otherName` could be freely extended and the quality of its encoding might not be as guaranteed as the other ones. @seanjmullan, any comment here?

Too many "other" used and hopefully no one get confused.

Reply:
This seems like a reasonable balance between preserving existing behavior and still providing the data for the application to inspect. To help debugging, you could log the exception.